vms/volunteer/utils.py

```diff
@@ -0,0 +1,19 @@
+from functools import wraps
+from django.shortcuts import render
+from django.http import Http404
+from volunteer.services import get_volunteer_by_id
+
+def vol_id_check(func):
+    @wraps(func)
+    def wrapped_view(request, volunteer_id):
+        vol = getattr(request.user, 'volunteer', hasattr(request.user, 'administrator'))
+        if not vol:
```

Comment:
Yes, but also if the user was added through the admin interface, where they aren't an admin or a volunteer, it will display a no rights page